### PR TITLE
[BIOMAGE-1506] Fix 2 small things in continuous embedding plot

### PR DESCRIPTION
**Pull request by cosa65**
# Background
#### Link to issue 
https://biomage.atlassian.net/browse/BIOMAGE-1506

#### Link to staging deployment URL 
https://ui-martinfosco-u583.scp-staging.biomage.net/

#### Anything else the reviewers should know about the changes here

# Changes
### Code changes

# Definition of DONE
Your changes will be ready for merging after each of the steps below have been completed:

### Code updates
Have best practices and ongoing refactors being observed in this PR

- [ ] Migrated any selector / reducer used to the new format
- [ ] Validated that current unit tests work as expected and are enough

### Testing
- [ ] Unit tests written
- [ ] Tested locally with Inframock (with latest production data downloaded with biomage experiment pull)
- [ ] Deployed to staging
- [ ] Passed integration tests 

To set up easy local testing with inframock, follow the instructions here: https://github.com/biomage-ltd/inframock
To deploy to the staging environment, follow the instructions here: https://github.com/biomage-ltd/biomage-utils

### Documentation updates
Is all relevant documentation updated to reflect the proposed changes in this PR?

- [ ] Relevant Github READMEs updated
- [ ] Relevant wiki pages created/updated

### Approvers
- [ ] Approved by a member of the core engineering team
- [ ] (UX changes) Approved by vickymorrison (this is her username, tag her if you need approval)

### Just before merging:
- [ ] After the PR is approved, the `unstage` script in here: https://github.com/biomage-ltd/biomage-utils is executed. This script cleans up your deployment to staging

### Optional
- [ ] Photo of a cute animal attached to this PR

### CODE REVIEW

File: src/redux/reducers/componentConfig/initialState.js
Comment (cosa65) [inline src/redux/reducers/componentConfig/initialState.js:74]: I replaced this `notSelected` value with `null` because it will never be seen: if the component detects this value it shows loading until it displays the default gene. (The only reason it was being seen was a bug)

The value was performing the same function that `null` usually does only it wasn't `null`.

File: src/components/plots/styling/SingleGeneSelection.jsx
Comment (StefanBabukov) [inline src/components/plots/styling/SingleGeneSelection.jsx]: I think if you just leave `!config?.shownGene` in the if it should be enough?

File: src/__test__/components/plots/styling/SingleGeneSelection.test.jsx
Comment (StefanBabukov) [inline src/__test__/components/plots/styling/SingleGeneSelection.test.jsx:13]: Why do you need all this act wraps everywhere, when I write tests I dont use them and I don't get errors/warnings?

File: src/__test__/components/plots/styling/SingleGeneSelection.test.jsx
Comment (StefanBabukov) [inline src/__test__/components/plots/styling/SingleGeneSelection.test.jsx]: Why do you wrap some of the `loadAndRenderSingleGeneSelection` calls in act and others not?

File: src/__test__/components/plots/styling/SingleGeneSelection.test.jsx
Comment (StefanBabukov) [reply src/__test__/components/plots/styling/SingleGeneSelection.test.jsx:13]: I thought they were needed for the enzyme tests

File: src/__test__/data/componentConfigs/embeddingContinuous.json
Comment (StefanBabukov) [inline src/__test__/data/componentConfigs/embeddingContinuous.json:1]: nice, the spec file was getting out of hand :D

File: src/__test__/components/plots/ContinuousEmbeddingPlot.test.jsx
Comment (StefanBabukov) [inline src/__test__/components/plots/ContinuousEmbeddingPlot.test.jsx]: Maybe you can put `mockedStore.genes.expression` into a variable, so these props across the file are shorter

File: src/__test__/components/plots/styling/SingleGeneSelection.test.jsx
Comment (cosa65) [reply src/__test__/components/plots/styling/SingleGeneSelection.test.jsx:13]: I think it is a react testing thing, when you perform actions that should cause re renderings wrapping these actions in act ensures the updates are done. https://reactjs.org/docs/testing-recipes.html#act

File: src/__test__/components/plots/styling/SingleGeneSelection.test.jsx
Comment (cosa65) [reply src/__test__/components/plots/styling/SingleGeneSelection.test.jsx]: oh, I shouldn't wrap those in `act`, inside them there is an `act`

File: src/redux/reducers/componentConfig/initialState.js
Comment (kafkasl) [inline src/redux/reducers/componentConfig/initialState.js:106]: why this change?

File: src/pages/experiments/[experimentId]/plots-and-tables/embedding-continuous/index.jsx
Comment (kafkasl) [inline src/pages/experiments/[experimentId]/plots-and-tables/embedding-continuous/index.jsx]: this comment does not clarify what's happening to me. "If there is a saved gene in the config" I assume refers to `config?.shownGene` but then what about the `!searchedGene`? Could you reword this to make it more clear?

also I'd probably reorder the if so the `?` is not needed: `config && config.shownGene !== null && !searchedGene` but I leave it to your decision

File: src/redux/reducers/componentConfig/initialState.js
Comment (kafkasl) [inline src/redux/reducers/componentConfig/initialState.js:76]: I assume this fixes the capped / uncapped issue but it's not very clear to me how. Also, truncated / capped are equivalent names? it's a bit confusing

**Review by kafkasl — APPROVED** [review]

File: src/redux/reducers/componentConfig/initialState.js
Comment (cosa65) [reply src/redux/reducers/componentConfig/initialState.js:76]: `expressionValue` and `truncatedValues` were being searched for but didn't exist in the initial config of this plot, so what I did to fix that issue was add them to initial config

File: src/redux/reducers/componentConfig/initialState.js
Comment (cosa65) [reply src/redux/reducers/componentConfig/initialState.js:106]: because the default coloring values for heatmap are supposed to be the truncated expressions, I double checked with @ogibson-biomage  to be sure

File: src/pages/experiments/[experimentId]/plots-and-tables/embedding-continuous/index.jsx
Comment (cosa65) [reply src/pages/experiments/[experimentId]/plots-and-tables/embedding-continuous/index.jsx]: We don't need to check `config` at all, just doing `config?.shownGene !== null` handles config being `null` too.

About the comment, I took a brief look at what it is doing and updated the comment, but based on what I saw this could benefit from a more integral refactoring probably to make `searchedGene`, `shownGene` and the expressions loading a little bit more cohesive. This PR is about fixing the bugs though so I am not sure if perhaps I shoudl create a ticket for that to avoid holding this up.

Comment (kafkasl) [reply src/pages/experiments/[experimentId]/plots-and-tables/embedding-continuous/index.jsx]: I just meant to improve the comment which you did so now it's clear, thanks; and yeah the `config` can be removed alltogether.

**Comment by vickymorrison** [comment]
In staging, the first time I go to the continuous embedding plot I get "it's not you, it's us". Then if I go back to Plots & Tables home page and then click the continuous embedding plot again, it loads fine. I tried with 3 different experiments - the same thing happened every time. Video below (the plot did load I just didn't record it). 
https://user-images.githubusercontent.com/74908153/146605416-5d91b376-2ad3-4483-af4a-b75d541a3384.mov

The 2 issues are fixed though!

**Comment by cosa65** [comment]
> In staging, the first time I go to the continuous embedding plot I get "it's not you, it's us". Then if I go back to Plots & Tables home page and then click the continuous embedding plot again, it loads fine. I tried with 3 different experiments - the same thing happened every time. Video below (the plot did load I just didn't record it). https://user-images.githubusercontent.com/74908153/146605416-5d91b376-2ad3-4483-af4a-b75d541a3384.mov
> 
> The 2 issues are fixed though!

Thanks! I'll take a look

**Review by vickymorrison — APPROVED** [review]

**Pull request merged**
